esp32s2-hal/ld/memory.x

```diff
@@ -21,17 +23,17 @@ STACK_SIZE = 8k;
 /* Specify main memory areas */
 MEMORY
 {
-  vectors_seg ( RX )     : ORIGIN = 0x40022000, len = 1k /* SRAM0 */
-  iram_seg ( RX )        : ORIGIN = 0x40022400, len = 128k-0x400 /* SRAM0 */
+  vectors_seg ( RX )     : ORIGIN = 0x40020000 + RESERVE_DRAM, len = VECTORS_SIZE /* SRAM0 */
+  iram_seg ( RX )        : ORIGIN = 0x40020000 + RESERVE_DRAM + VECTORS_SIZE, len = 192 - RESERVE_DRAM - VECTORS_SIZE /* SRAM0 */
 
-  dram_seg ( RW )        : ORIGIN = 0x3FFB0000 + RESERVE_DRAM, len = 192k - RESERVE_DRAM
+  dram_seg ( RW )        : ORIGIN = 0x3FFB0000 + RESERVE_DRAM + VECTORS_SIZE, len = 192k - RESERVE_DRAM - VECTORS_SIZE
 
   /* SRAM1; reserved for static ROM usage; can be used for heap.
      Length based on the "_dram0_rtos_reserved_start" symbol from IDF used to delimit the
      ROM data reserved region:
      https://github.com/espressif/esp-idf/blob/bcb34ca7aef4e8d3b97d75ad069b960fb1c17c16/components/heap/port/esp32s2/memory_layout.c#L121-L122
   */
-  reserved_for_boot_seg  : ORIGIN = 0x3FFE0000, len = 0x1FA10
+  reserved_for_boot_seg  : ORIGIN = 0x3ffffa10, len = 0x5f0
```

Comment:
I didn't understand this change here, could you please explain the new origin and length values?

Reply:
I got this from https://github.com/espressif/esp-idf/blob/2b1bee166d4d5b55549063e93a4c1be86e9c9da3/components/esp_rom/esp32s2/ld/esp32s2.rom.ld#L194

which is mentioned in the link contained in the comment

but to be honest I am not sure about that ... will just revert it to the previous values

Reply:
Probably the idea was similar here but this doesn't look right:
`_heap_end = ABSOLUTE(ORIGIN(dram_seg))+LENGTH(dram_seg)+LENGTH(reserved_for_boot_seg) - STACK_SIZE;`

I think it should be `-LENGTH(reserved_for_boot_seg)`

Reply:
The `reserved_for_boot_seg` is an area still in DRAM, but not considered in the `dram_seg` because it is used by the Bootloader for actually loading the Application Image code and data into IRAM and DRAM. Otherwise the Bootloader could end up overwriting itself and failing to load the Application Image properly.

After the booting process is finished, the Application Image may reclaim the IRAM area used by Bootloader, this area is represented by `reserved_for_boot_seg`. So I believe the calculus is correct.